Composer/packages/tools/language-servers/language-generation/src/utils.ts

```diff
@@ -153,6 +153,20 @@ export const cardTypes = [
   'Activity',
 ];
 
+export const cardPropPossibleValueType = {
+  title: 'An Example Card',
+  type: 'Action Type',
+  value: 'Some Value',
+  SuggestionActions: 'Text | ${Some_CardAction}',
+  subtitle: 'An Example Subtitle',
+  text: 'Some text',
+  image: 'https://example.com/demo.jpg',
+  buttons: 'Text | ${Some_CardAction}',
+  contenttype: 'adaptivecard',
+  content: '${json(fromFile("../../card.json"))}',
+  name: 'An Example Name',
+};
+
```

Comment:
Can these use the vs code placeholder symantics?

For example: `text: '${1:Some text}'

See the example below of a snippet that does this.
![Screen Recording 2020-04-16 at 09 19 AM](https://user-images.githubusercontent.com/3619060/79480752-5df66300-7fc3-11ea-987c-f33259353d5c.gif)

Reply:
I tried using snippet string with placeholder during adding suggestion of builtin functions. However, the completionItem in LSP cannot do the placeholder directly. I think it needs some transformation in monaco client side. Furthermore, the snippet ${1: some text} looks confusing in the context of we can using expression as a value here.

Reply:
I see. This aren't really "snippets" in the vs code sense.